htmleditor-mixin.js

```diff
@@ -739,19 +740,76 @@ export const HtmlEditorMixin = superclass => class extends Localizer(LabelledMix
 					editor.undoManager.clear();
 				});
 
-				const persistEditorContents = () => {
-					if (this.pasteLocalImages) editor.uploadImages();
-					this._html = editor.getContent();
+				let persistEditorContents = () => {
+					if (this.pasteLocalImages) {
+						editor.uploadImages();
+					}
+					this.html = editor.getContent();
 				};
 
-				editor.on('detach', () => {
+				let editorDetach = () => {
 					persistEditorContents();
-				});
+				};
 
-				editor.on('focusout', () => {
+				let editorFocusOut = () => {
 					persistEditorContents();
 					this._handleFocusOut();
-				});
+				};
+
+				if (this._context && this._context.useNewTinyMCEVersion) {
+					persistEditorContents = async() => {
+						const scannedImages = await editor.editorUpload.scanForImages();
+						const blobCache = editor.editorUpload.blobCache;
+
+						scannedImages.forEach(scannedImage => {
+							const scannedBlobInfo = scannedImage.blobInfo;
+							if (scannedBlobInfo.filename().startsWith('editimage') === false) return;
```

Comment:
```suggestion
							if (!scannedBlobInfo.filename().startsWith('editimage')) return;
```
No need for strict equality here.